src/components/UploadAvatarModal/UploadAvatarModal.js

```diff
@@ -0,0 +1,247 @@
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import PropTypes from 'prop-types'
+import Slider from 'rc-slider'
+import { useState, useCallback } from 'react'
+import Cropper from 'react-easy-crop'
+import { useDispatch } from 'react-redux'
+
+import asModal, { ModalContent, ModalFooter } from '~/components/asModal'
+import { updateAvatar } from '~/store/actions/user'
+import getResponseError from '~/util/getResponseError'
+
+import UploadAvatarMessageBox from './UploadAvatarMessageBox'
+import styles from './UploadAvatarModal.module.scss'
+
+// Component Constants
+const MAX_FILE_SIZE = 26214400 // Server upload max is 25M
+const SUBMIT_AUTO_CLOSE_DELAY_TIME = 3000
+const HALF_CIRCLE = 180 // Conversion of rat to deg
+
+
+function UploadAvatarModal (props) {
+  const {
+    onClose,
+    isOpen,
+  } = props
+
+  const [result, setResult] = useState({})
+
+  const [upImg, setUpImg] = useState()
+  const [crop, setCrop] = useState({ x: 0, y: 0 }) // Lint comment on "Identifier name is too short" - this is required by the Cropper. Cannot change.
```

Comment:
Please do not leave lint comments like this. Linting is enforced by our CI, and will refuse to deploy when left unresolved. if something like this MUST remain (in this case it does), then use a `eslint-disable-next-line` directive for the reporting rule.

```suggestion
  // eslint-disable-next-line id-length -- Required by react-easy-crop
  const [crop, setCrop] = useState({ x: 0, y: 0 })
```